required.go

```diff
@@ -6,35 +6,73 @@ import (
 	"strings"
 )
 
+type Option uint8
+
+const (
+	// Allows empty string, array or map. Default is false
+	AllowEmptyString Option = iota
```

Comment:
To see a more idiomatic way of doing this, check out the `encoding/csv` package's CSV reader. Basically, these should be fields on an instance of your `Required` validator.